File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/ui/itemview/TransactionItemComponent.java
package com.fintechplatform.ui.transactions.ui.itemview;

import com.fintechplatform.ui.alert.AlertHelpersModule;
import com.fintechplatform.ui.money.MoneyHelperModule;
import com.fintechplatform.ui.transactions.ui.TransactionItemViewHolder;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {TransactionItemPresenterModule.class,
        AlertHelpersModule.class,
        MoneyHelperModule.class})
public interface TransactionItemComponent {

    void inject(TransactionItemViewHolder viewHolder);
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/di/TransactionItemComponent.kt
package com.fintechplatform.ui.transactions.di

import com.fintechplatform.ui.alert.AlertHelpersModule
import com.fintechplatform.ui.money.MoneyHelperModule
import com.fintechplatform.ui.transactions.ui.itemview.TransactionItemViewHolder
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [TransactionItemPresenterModule::class, AlertHelpersModule::class, MoneyHelperModule::class])
interface TransactionItemComponent {

    fun inject(viewHolder: TransactionItemViewHolder)
}
